File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    static Set<Integer> nums = new HashSet<>();
    static boolean[] visited;
    
    public int solution(String numbers) {
        int answer = 0;
        visited = new boolean[numbers.length()];
        permutations("", numbers);
        
        for (int n: nums) {
            if (isPrime(n)) {
                answer++;
            }
        }
        return answer;
    }
    
    // 순열 조합 만드는 메서드
    void permutations(String current, String numbers) {
        if (!current.equals("")) { // 현재 만든 숫자(current) 저장
            nums.add(Integer.parseInt(current));
        }
        
        for (int i = 0; i < numbers.length(); i++) {
            if (!visited[i]) {
                visited[i] = true;
                permutations(current + numbers.charAt(i), numbers); // current에 다음 글자 추가
                visited[i] = false; // 백트래킹
            }
        }
    }
    
    boolean isPrime(int n) {
        if (n < 2) return false;
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) return false;
        }
        return true;
    }
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def is_prime(n):
    if n < 2:
        return False
    
    for i in range(2, int(n**(1/2))+1):
        if n % i == 0:
            return False
    return True

def solution(numbers):
    answer = 0
    
    nums = set()
    for i in range(1, len(numbers)+1):
        for p in permutations(numbers, i):
            n = int(''.join(p))
            nums.add(n)
    
    for num in nums:
        if is_prime(num):
            answer += 1
    return answer